Baikal/PostEffects/ML/buffer.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include <vector>
+#include <cstddef>
+#include <memory>
+
+
+namespace Baikal
+{
+    namespace PostEffects
+    {
+        class Buffer
+        {
+        public:
+            using ValueType = float;
+            using Data = std::unique_ptr<ValueType[]>;
+            using Shape = std::tupel<size_t, size_t, size_t>;
+
+            Buffer(Data data, Shape shape) :
+                m_data(std::move(data)),
+                m_shape(std::move(shape)),
+                m_size(std::get<0>(m_shape) *
+                       std::get<1>(m_shape) *
+                       std::get<2>(m_shape))
+            {
+            }
+
+            bool empty() const
```

Comment:
I don't like this empty(), maybe use std::optional? Let's do this as separate project just like BaikalDataGenerator and use C++17

Reply:
Basically I agree. But we did not discuss where to put the new post effect code.

Reply:
No, we'll not create dedicated project. We'll use C++ 14 instead of this